src/rust/lib/graph-editor/src/component/operator.rs

```diff
@@ -0,0 +1,271 @@
+//! able to access some data about them and manipulate them in a consistent manner. For example:
+//! instead of every component implementing a "fullscreen mode", they just provide resizing
+//! functionality and access to their su-components and instead of determining the layer of shapes
+//! within the UI component, we instead make the shapes accessible via an API that indicates the
+//! expected behaviour of the shapes, and the actual layer management happens outside of the
+//! component.
+//!
+//! This is also a step to avoid relying too much on the presence of the `Scene` within the UI
+//! component.
+use crate::prelude::*;
+
+use ensogl::display::DomScene;
+use ensogl::display::DomSymbol;
+use ensogl::display::Scene;
+use ensogl::display::Symbol;
+use ensogl::display::scene::View;
+use ensogl::display::traits::*;
+use ensogl::display;
+use ensogl::frp;
+use ensogl::gui::component::animation;
+
+
+
+// ==================================
+// === UI Component Helper Traits ===
+// ==================================
+
+/// Should be implemented by UI component that consist of `Symbol`. Provides access to the shapes
+/// and some helper methods for working with those shapes.
+pub trait NativeComponent {
+    /// Return all `Symbol`s that make up this component.
+    fn symbols(&self) -> Vec<SymbolType>;
+
+    /// Change the scene layer of all `Symbol`s.
+    fn set_layer(&self, layer:&View) {
+        self.symbols().iter().for_each(|symbol| {
+            match symbol{
+                SymbolType::Main(symbol)
+                | SymbolType::Visualisation(symbol)
+                => layer.add(symbol),
+            }
+        })
+    }
+
+    /// Remove the `Symbol`s from all scene layers.
+    fn unset_layers_all(&self, scene:&Scene) {
+        self.symbols().iter().for_each(|symbol|   match symbol{
+            SymbolType::Main(symbol)
+            | SymbolType::Visualisation(symbol)
+            =>scene.views.remove_symbol(symbol),
+        })
+    }
+}
+
+/// Should be implemented by UI component that consist of `DomSymbol`s. Provides access to the
+/// symbols some helper methods for working with them.
+pub trait HtmlComponent {
+    /// Return all `DomSymbol`s that make up this component.
+    fn elements(&self) -> Vec<DomSymbol>;
+
+    /// Change the `DomScene` of all `DomSymbol`s.
+    fn set_layer(&self, scene:&DomScene) {
+        self.elements().iter().for_each(|element| scene.manage(&element))
+    }
+}
+
+/// Provides information and functionality to resize an element. A complex UI component should
+/// implement this and propagate size and layout changes to all its sub-components.
+pub trait Resizable {
+    /// Set the size for the UI component.
+    fn set_size(&self, size:Vector3<f32>);
+    /// Return the size of the UI element.
+    fn size(&self) -> Vector3<f32>;
+}
+
+/// A component that owns a Network. Can be used to create external animations.
+pub trait Networked {
```

Comment:
We have naming convention for that - it should be named `HasNetwork` (and should be placed in `frp::` space). BUT, it was not defined on purpose. I believe that defining it encourages you to actually leak memory accidently. See comments below. You should not access frp network on demand - only when initializing an object. Otherwise its too easy to leak memory. I would remove it.